presto-native-execution/presto_cpp/main/types/VeloxToPrestoExpr.h

```diff
@@ -0,0 +1,108 @@
+/// equivalent Presto expression of type `protocol::RowExpression`:
+///  1. A constant Velox expression of type `core::ConstantTypedExpr` is
+///     converted to a Presto expression of type `protocol::ConstantExpression`.
+///     If the Velox constant expression is of `ROW` type, it is converted to a
+///     Presto expression of type `protocol::SpecialFormExpression` with Form
+///     as `ROW_CONSTRUCTOR`.
+///  2. A Velox expression representing a field in the input, of type
+///     `core::FieldAccessTypedExpr`, is converted to a Presto expression of
+///     type `protocol::VariableReferenceExpression`.
+///  3. A Velox dereference expression of type `core::DereferenceTypedExpr` is
+///     converted to a Presto expression of type
+///     `protocol::SpecialFormExpression` with Form as `DEREFERENCE`.
+///  4. A Velox cast expression of type `core::CastTypedExpr` is converted to a
+///     Presto expression of type `protocol::CallExpression`.
+///  5. A Velox call expression of type `core::CallTypedExpr` is converted
+///     either to a Presto expression of type `protocol::CallExpression` or of
+///     type `protocol::SpecialFormExpression`. This is because Velox does
+///     not have a separate expression kind for special form expressions and the
+///     special forms in Presto and Velox do not have a one to one mapping. If
+///     the velox call expression's name belongs to the set of Presto's
+///     `SpecialFormExpression` names, it is converted to a Presto
+///     `protocol::SpecialFormExpression`; else it is converted to a Presto
+///     `protocol::CallExpression`.
+class VeloxToPrestoExprConverter {
+ public:
+  explicit VeloxToPrestoExprConverter(velox::memory::MemoryPool* pool)
+      : pool_(pool) {}
+
+  /// Converts a Velox expression `expr` to a Presto protocol RowExpression.
+  /// The input Presto protocol RowExpression, `inputRowExpr`, is returned in
+  /// case the Velox to Presto expression conversion fails.
+  RowExpressionPtr getRowExpression(
+      const velox::core::TypedExprPtr& expr,
+      const RowExpressionPtr& inputRowExpr = nullptr) const;
+
+ private:
+  /// `ValueBlock` in Presto `protocol::ConstantExpression` requires only the
+  /// column from the serialized PrestoPage without the page header. This
+  /// function is used to serialize a velox vector to `ValueBlock`.
+  std::string getValueBlock(const velox::VectorPtr& vector) const;
+
+  /// Helper function to get a Presto `protocol::ConstantExpression` from a
+  /// Velox constant expression.
+  ConstantExpressionPtr getConstantExpression(
+      const velox::core::ConstantTypedExpr* constantExpr) const;
+
+  /// Helper function to get the arguments for Presto `SWITCH` expression from
+  /// Velox switch expression.
+  std::vector<RowExpressionPtr> getSwitchSpecialFormExpressionArgs(
+      const velox::core::CallTypedExpr* switchExpr) const;
+
+  /// Helper function to construct a Presto `protocol::SpecialFormExpression`
```

Comment:
What determines if the Call is a SpecialFormExpression. Please give more details here.